### PR TITLE
unnecessary_stateful_widgets

### PR DESCRIPTION
**Pull request by a14n**
This PR fixes diagnostics from https://github.com/dart-lang/linter/pull/3725 to catch StatefulWidget that can be changed to StatelessWidget

### CODE REVIEW

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 1]
```diff
@@ -519,10 +514,10 @@ class _ReorderableListViewState extends State<ReorderableListView> {
 // of the objects used to generate widgets.
 @optionalTypeArgs
 class _ReorderableListViewChildGlobalKey extends GlobalObjectKey {
-  const _ReorderableListViewChildGlobalKey(this.subKey, this.state) : super(subKey);
+  const _ReorderableListViewChildGlobalKey(this.subKey, this.context) : super(subKey);
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/reorderable_list.dart:517]: This is a little iffy. We tell people to not hold on to context objects like that.

File: packages/flutter/lib/src/cupertino/page_scaffold.dart [hunk 2]
```diff
@@ -196,7 +190,7 @@ class _CupertinoPageScaffoldState extends State<CupertinoPageScaffold> {
             height: existingMediaQuery.padding.top,
             child: GestureDetector(
               excludeFromSemantics: true,
-              onTap: _handleStatusBarTap,
+              onTap: () => _handleStatusBarTap(context),
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/src/cupertino/page_scaffold.dart:193]: (this also applies to some other cases in this PR)

**Comment by goderbauer** [comment]
> Do you have any idea why the tests fail after converting them to StatelessWidgets?

I briefly looked at this and it isn't obvious to me why this is happening...

File: packages/flutter/lib/src/cupertino/page_scaffold.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (a14n) [reply packages/flutter/lib/src/cupertino/page_scaffold.dart:193]: I updated the lint to have an exception if the WidgetState use a member of its parent.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**